### PR TITLE
Added release notes for self-signed certificates

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Adds release notes for self-signed certificates

### What issues does this PR fix or reference?
https://github.com/eclipse/che/pull/12089

### CODE REVIEW

**Comment by sleshchenko** [comment]
@slemeur @eivantsov @rkratky release notes template is confusing for me. We don't have any release document that is based on it as an example (at least I didn't find it easily). So, I might do it wrong. Please review/comment/suggest changes.

**Comment by rkratky** [comment]
CC @dneary

**Comment by rkratky** [comment]
@sleshchenko Here's the published doc for 6.14: https://che.eclipse.org/eclipse-che-6-14-release-notes-971dd36d23ed

**Comment by sleshchenko** [comment]
@rkratky Thanks for sharing. 
If I understand correctly there is someone (like Dave) who gets release notes templates made by feature developer and then improve it, fill up changes that are described in PR but not in a template and publish to Che Docs Site.
I see several issues here: 
1. PR template in Che Repository [[1]](https://raw.githubusercontent.com/eclipse/che/master/.github/PULL_REQUEST_TEMPLATE.md) does not have any mentions that there is any template file for releases notes that should be filled in.

2. Who is supposed to create a new template file for each release with $VERSION, $RELEASE filled in? If a person who made the first PR which require release notes - then there may be extra work that may be done by two persons who made such PRs at the same time. Also, there are possible issues with resolving git conflicts since each feature.

I think it makes sense to rethink our release notes writing process to provide a more comfortable and clear way for features authors. First of all, Github template should be up to date. 
Then I think that it may be easier for features authors to have the following structure of release notes documents:
```
release-notes\
   | - 6.15.0\
   |          | - cors-disabling-feature.md
   |          | - self-signed-certificates.md
   | - rn-template.md // <- needed for Release notes author but not feature author
   | - feature-notes-template.md // <- needed for Feature author
```
@rkratky @dneary @eivantsov @slemeur WDYT?

[1] https://raw.githubusercontent.com/eclipse/che/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by rkratky** [comment]
> PR template in Che Repository [1] does not have any mentions that there is any template file for releases notes that should be filled in.

I wasn't aware of that, but it can be easily fixed.

> Who is supposed to create a new template file for each release with $VERSION, $RELEASE filled in? If a person who made the first PR which require release notes - then there may be extra work that may be done by two persons who made such PRs at the same time.

That would be @dneary. I believe the hiccup here is that Dave is not around until the new year. Otherwise, I hope this won't cause problems.

> Also, there are possible issues with resolving git conflicts since each feature.

The idea is for every new PR submitter to copy the new-feature part of the template and populate it with their submission. Such workflow should result in conflicts hopefully only rarely. And if it does, then Dave should be able to easily resolve them when merging the PRs (I have purposefully refrained from merging the RN PRs because I didn't want to mess up whatever system Dave had.)

>  Then I think that it may be easier for features authors to have the following structure of release notes documents...

Looks good to me, but let's wait for Dave to have a say.

File: release-notes/rn-6.16.0.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* Kubernetes ([single-user](https://www.eclipse.org/che/docs/kubernetes-single-user.html) or [multi-user](https://www.eclipse.org/che/docs/kubernetes-multi-user.html))
+* OpenShift ([single-user](https://www.eclipse.org/che/docs/openshift-single-user.html) or [multi-user](https://www.eclipse.org/che/docs/openshift-multi-user.html))
+* Docker ([single-user](https://www.eclipse.org/che/docs/docker-single-user.html) or [multi-user](https://www.eclipse.org/che/docs/docker-multi-user.html))
+* Try Eclipse Che live at [https://che.openshift.io](https://che.openshift.io)
+
+Learn more in our [documentation](https://www.eclipse.org/che/docs/infra-support.html) and start using a shared Che server or local instance today.
+
+---
+
+## An ability to run Che Server with K8s/OS infrastructure on https with a self-signed certificate
```
Comment (dneary) [inline release-notes/rn-6.16.0.md]: Change to "Run Eclipse Che over https with a self-signed certificate"

File: release-notes/rn-6.16.0.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+![Che_on_https](https://www.eclipse.org/che/docs/images/workspaces/chrome_cert.png)
+
```
Comment (dneary) [inline release-notes/rn-6.16.0.md:39]: I suggest adding a description of why this is a good thing.

"Security is increasingly important for web services, and many Eclipse Che users would like to ensure communication between the Che server and client is encrypted and authenticated. In previous versions, it was not possible to start workspaces in Eclipse Che if your container platform was using a self-signed certificate. In this release, we have added that ability. The Che administrator can enable this feature by creating a secret with the certificate, and configuring the server to ensure that the certificate is used by the Che server.

For more information, see  https://www.eclipse.org/che/docs/che-6/openshift-config.html#https-mode---self-signed-certs"

I can't really tell from the docs and from this description is this is an easy feature to use - the docs say that I have to enable the certificate for each workspace, but I don't know how to do that. Would it be possible (in the docs) to clarify how to use this feature?

**Review by dneary — COMMENTED** [review]
Can you update this PR to take the following comments into account, please?

**Comment by sleshchenko** [comment]
@dneary Can you comment the proposed RH pages structure? WDYT about it?

File: release-notes/rn-6.16.0.md (hunk 2 above)
Comment (sleshchenko) [reply release-notes/rn-6.16.0.md:39]: > I suggest adding a description of why this is a good thing.

Done.

> I can't really tell from the docs and from this description is this is an easy feature to use - the docs say that I have to enable the certificate for each workspace, but I don't know how to do that. Would it be possible (in the docs) to clarify how to use this feature?

I think https://www.eclipse.org/che/docs/che-6/openshift-config.html#https-mode---self-signed-certs is not up to date. Have you seen Eugene's PR? Maybe things are clearer there https://github.com/eclipse/che-docs/pull/630/files
Please notify me if changes according to this comment are still needed.

File: release-notes/rn-6.16.0.md (hunk 1 above)
Comment (sleshchenko) [reply release-notes/rn-6.16.0.md]: Done

**Comment by dneary** [comment]
> @dneary Can you comment the proposed RH pages structure? WDYT about it?

Apologies - was just running through PR backlog, did not see your proposal.
The main difficulties for me are when the release note proposals don't match an aspect of what we need (clear user impact and explanation of how to use it, aggregating changes that are similar into a single section/theme, for example).

If we get review times down, it should be straightforward to use a single file - I don't see any reason to split things across multiple files, it will just make things more complicated for me.

**Comment by skabashnyuk** [comment]
@rkratky @dneary @slemeur can you review this pr. @sleshchenko has fixed last comments

**Review by slemeur — APPROVED** [review]
good for me as long @dneary is fine with the changes

**Pull request merged**